File: 2017 COMLAN/RangeValueMonitor/src-weaved/matlab/foobar.m
function foobar()
   monitor_range_init();
   foo();
   monitor_range_print();
end

function [a] = foo()
   % monitoring a
   a = 0;
   monitor_range_update(1, a);
   for i = 0:5
      % monitoring a
      a = a + bar();
      monitor_range_update(1, a);
   end
   % monitoring a
   a = libcall(a);
   monitor_range_update(1, a);
end

function [result] = bar()
   % monitoring result
   result = 1.0;
   monitor_range_update(2, result);
end

File: 2017 COMLAN/RangeValueMonitor/src-weaved/java/FooBar.java



public class FooBar {
    public double bar() {
        return 1.0;
    }

    public double foo() {
        double a = 0;
        kadabra.utils.RangeMonitors.monitor_range_update(0, a);
        for (int i = 0; i < 5; i++) {
            a += bar();
            kadabra.utils.RangeMonitors.monitor_range_update(0, a);
        }
        return a;
    }

    public static void main(String[] args) {
        new FooBar().foo();
        kadabra.utils.RangeMonitors.monitor_range_print();
    }
}

